connaisseur/config.py

```diff
@@ -27,55 +31,72 @@ def __init__(self):
         with open(self.__PATH, "r") as configfile:
             config_content = yaml.safe_load(configfile)
 
-        if not config_content:
+        with open(self.__SECRETS_PATH, "r") as secrets_configfile:
+            secrets_config_content = yaml.safe_load(secrets_configfile)
+
+        config = self.__merge_configs(config_content, secrets_config_content)
+
+        self.__validate(config)
+
+        self.validators = [Validator(**validator) for validator in config]
+
+    def __merge_configs(self, config: dict, secrets_config: dict):
+        if not config:
             msg = "Error loading connaisseur config file."
             raise NotFoundException(message=msg)
 
-        self.__validate(config_content)
-
-        self.notaries = [Notary(**notary) for notary in config_content]
+        for validator in config:
+            validator.update(secrets_config.get(validator.get("name"), {}))
+            auth_path = f'{self.__EXTERNAL_PATH}{validator["name"]}/auth.yaml'
+            if safe_path_func(os.path.exists, self.__EXTERNAL_PATH, auth_path):
+                with safe_path_func(
+                    open, self.__EXTERNAL_PATH, auth_path, "r"
+                ) as auth_file:
+                    auth_dict = {"auth": yaml.safe_load(auth_file)}
+                validator.update(auth_dict)
+        return config
 
     def __validate(self, config: dict):
-
         validate_schema(
             config,
             self.__SCHEMA_PATH,
             "Connaisseur configuration",
             InvalidConfigurationFormatError,
         )
-        notary_names = [notary.get("name") for notary in config]
-        if collections.Counter(notary_names)["default"] > 1:
-            msg = "Too many default notary configurations."
+        validator_names = [validator.get("name") for validator in config]
+        if collections.Counter(validator_names)["default"] > 1:
+            msg = "Too many default validator configurations."
             raise InvalidConfigurationFormatError(message=msg)
 
-        for notary in config:
-            key_names = [key.get("name") for key in notary.get("pub_root_keys")]
-            if collections.Counter(key_names)["default"] > 1:
-                msg = "Too many default keys in notary configuration {notary_name}."
-                raise InvalidConfigurationFormatError(
-                    message=msg, notary_name=notary.get("name")
-                )
-
-        self.notaries = [Notary(**notary) for notary in config]
+        # for notary in config:
+        #     key_names = [key.get("name") for key in notary.get("pub_root_keys")]
+        #     if collections.Counter(key_names)["default"] > 1:
+        #         msg = "Too many default keys in notary configuration {notary_name}."
+        #         raise InvalidConfigurationFormatError(
+        #             message=msg, notary_name=notary.get("name")
+        #         )
```

Comment:
was this forgotten?

Reply:
eh...that doesn't work for the static validators, who don't have any keys, so i temporarly took it out. I was think about putting the validation of validator-specific things into the respective validators, but haven't done so yet. then the validation of the config file will just check whether there is a `name` and `type` for a validator (+ only one is called default) and everything else is done by the validators themselves. but this will be probably done in a seperate PR :)

Reply:
That sounds reasonable. But shouldn't this orphaned comment be deleted, then? (If there's a to-do, let's make it an issue)